frontend/src/tests/posting-tests.js

```diff
@@ -1,39 +1,354 @@
+        resp = await apiPOST(
+            `/admin/postings/${posting.id}/posting_positions`,
+            { ...postingPos, position_id: position.id }
+        );
+        expect(resp).toHaveStatus("success");
+        checkPropTypes(postingPositionPropTypes, resp.payload);
+        expect(resp.payload.posting_id).toEqual(posting.id);
+        Object.assign(postingPos, resp.payload);
     });
 
-    it.todo("Create a posting for a session");
-    it.todo("Modify a posting");
-    it.todo("Cannot set the `status` of a posting to an invalid value");
-    it.todo("Fetch all postings for a session");
-    it.todo(
-        "Can set `custom_questions` to an arbitrary serializable object and the same object (i.e., an object and not the stringified version) gets returned."
-    );
-    it.todo("Two postings for the same session cannot have the same name");
-    it.todo("Two postings for different sessions may have the same name");
-    it.todo("Create a posting_position for a posting");
-    it.todo("A posting contains a list of all associated posting_position ids");
-    it.todo("A posting contains a list of all associated application ids");
-    it.todo("Fetch a survey for a posting");
-    it.todo(
-        "Survey for a posting includes questions related to each PostingPosition"
-    );
-    it.todo(
-        "Cannot create a posting_position with a position associated with a different session than the posting"
-    );
-    it.todo("Delete a posting_position");
-    it.todo("Modify a posting_position");
-    it.todo("Deleting a posting also deletes all associated posting_positions");
-    it.todo(
-        "Cannot create two posting_positions with the same position_id for a single posting"
-    );
-    it.todo(
-        "Can create two posting_positions with the same position_id for different postings"
-    );
+    // `/admin/postings/${posting.id}/applications` need to be implemented
+    it.todo("Fetch all applications associated with a posting");
+
+    it("Fetch a survey for a posting", async () => {
+        resp = await apiGET(`/admin/postings/${posting.id}/survey`);
+        expect(resp).toHaveStatus("success");
+    });
+
+    it("Survey for a posting includes questions related to each PostingPosition", async () => {
+        resp = await apiGET(`/admin/postings/${posting.id}/survey`);
+        expect(resp).toHaveStatus("success");
+        let survey = resp.payload;
+        let postingPosQuestions = survey.pages.filter(
+            (p) => p.name === "preferences_page"
+        )[0].elements[0].rows;
+        resp = await apiGET(`/admin/postings/${posting.id}/posting_positions`);
+        expect(resp).toHaveStatus("success");
+        expect(resp.payload.length).toEqual(postingPosQuestions.length);
+    });
+
+    it.skip("Cannot create a posting_position with a position associated with a different session than the posting", async () => {
+        const newSession = await addSession(api);
```

Comment:
you may only need just one another new `session` globally, and re-use it on this test as well as previous tests involving different session